src/Wellcome.Dds/Utils/Storage/FileSystem/FileSystemStorage.cs

```diff
@@ -3,68 +3,67 @@
 using System.IO;
 using System.Runtime.Serialization;
 using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading.Tasks;
 
 namespace Utils.Storage.FileSystem
 {
     public class FileSystemStorage : IStorage
     {
-        private ILogger<FileSystemStorage> logger;
+        private readonly ILogger<FileSystemStorage> logger;
 
         public FileSystemStorage(ILogger<FileSystemStorage> logger)
         {
             this.logger = logger;
         }
 
-        public string Folder { get; set; }
+        public string Container { get; set; }
 
         public ISimpleStoredFileInfo GetCachedFile(string fileName)
         {
-            string cachedFilePath = Path.Combine(Folder, fileName);
+            string cachedFilePath = Path.Combine(Container, fileName);
             FileInfo fi = new FileInfo(cachedFilePath);
             return new FileSystemStoredFileInfo(fi);
         }
         
         public void DeleteCacheFile(string fileName)
         {
-            string cachedFilePath = Path.Combine(Folder, fileName);
+            string cachedFilePath = Path.Combine(Container, fileName);
             File.Delete(cachedFilePath);
         }
 
-        public T Read<T>(ISimpleStoredFileInfo fileInfo) where T : class
+        public async Task<T> Read<T>(ISimpleStoredFileInfo fileInfo) where T : class
         {
+            if (!fileInfo.Exists) return null;
             T t = default(T);
-            if (fileInfo.Exists)
+            try
             {
-                try
+                IFormatter formatter = new BinaryFormatter();
+                await using (Stream stream = new FileStream(fileInfo.Uri, FileMode.Open, FileAccess.Read, FileShare.Read))
                 {
-                    IFormatter formatter = new BinaryFormatter();
-                    using (Stream stream = new FileStream(fileInfo.Uri, FileMode.Open, FileAccess.Read, FileShare.Read))
-                    {
-                        stream.Position = 0;
-                        t = formatter.Deserialize(stream) as T;
-                        stream.Close();
-                    }
-                    if (t == null)
-                    {
-                        logger.LogError("Attempt to deserialize '" + fileInfo.Uri + "' from disk failed.");
-                    }
+                    stream.Position = 0;
+                    t = formatter.Deserialize(stream) as T;
+                    stream.Close();
```

Comment:
These same lines are in `S3Storage`, worth pulling out? `t = formatter.DeserializeStream(stream)`?

```cs
stream.Position = 0;
t = formatter.Deserialize(stream) as T;
stream.Close();
```

Reply:
Not sure, I think the way it reads now is OK despite the duplication.